### ISSUE TITLE
upgrading to 3.0 usability feedback

### ISSUE THREAD

**Issue by dylang** [sindresorhus/trash#37]
Hi @sindresorhus, I ran into a small problem upgrading `trash` tonight. I thought I'd provide you what was going through my head in case you start hearing about issues from others who aren't as interesting in giving all of the details, and if you are interested in making changes to simplify the transition.

What happened:

*Hmm, I want to get rid of this pesky `node_modules` directory in my test fixtures.*

```
❯ trash node_modules


┌──────────────────────────────────────────┐
│ Update available: 3.0.0 (current: 2.0.0) │
│ Run npm install -g trash to update.      │
└──────────────────────────────────────────┘

```

*Fun, let's upgrade!*

```
 ❯ npm -g i trash
npm WARN deprecated xdg-trash@2.0.2: Deprecated in favor of the `trash` module
npm WARN deprecated osx-trash@2.0.0: Deprecated in favor of the `trash` module
npm WARN deprecated win-trash@2.0.0: Deprecated in favor of the `trash` module
trash@3.0.0 /Users/dgreene/.nvm/versions/node/v4.1.2/lib/node_modules/trash
├── win-trash@2.0.0 (pify@2.2.0, pinkie-promise@1.0.0)
├── osx-trash@2.0.0 (escape-string-applescript@1.0.0, run-applescript@2.0.0, pify@2.2.0, pinkie-promise@1.0.0)
└── xdg-trash@2.0.2 (pify@2.2.0, pinkie-promise@1.0.0, uuid@2.0.1, xdg-trashdir@2.0.0, fs-extra@0.18.4)
previous command took 5s.
```

*Well that's strange, I'm installing the latest version of `trash` but it's in turn installing node modules that say they are deprecated, and they want me instead to install `trash`, which is what I thought I was doing.* 

*Oh well, probably just a glitch in the matrix, these things happen.*

```
❯ trash node_modules
zsh: command not found: trash
```

*Hmm, what did __I__ break?*

After re-installing a few times, checking my version of Node, npm, trying `npm-check -g`, I run [changelog](https://www.npmjs.com/package/changelog) on `trash`:

```
❯ changelog trash
3.0.0 / 2015-10-13
  * 3.0.0
  * minify images further
  * extract CLI into a separate module
    https://github.com/sindresorhus/trash-cli
  * tweaks
  * Close https://github.com/sindresorhus/trash/issues/35 PR: Use promises and ES2015ify tests.
  * fix logo
```

*ooooh! new trash-cli. I think I saw that on twitter.*

```
❯ npm i -g trash-cli
npm WARN deprecated osx-trash@2.0.0: Deprecated in favor of the `trash` module
npm WARN deprecated xdg-trash@2.0.2: Deprecated in favor of the `trash` module
npm WARN deprecated win-trash@2.0.0: Deprecated in favor of the `trash` module
/Users/dgreene/.nvm/versions/node/v4.1.2/bin/trash -> /Users/dgreene/.nvm/versions/node/v4.1.2/lib/node_modules/trash-cli/cli.js
trash-cli@1.0.0 /Users/dgreene/.nvm/versions/node/v4.1.2/lib/node_modules/trash-cli
├── path-exists@2.0.0 (pinkie-promise@1.0.0)
├── meow@3.4.2 (loud-rejection@1.0.0, trim-newlines@1.0.0, object-assign@4.0.1, minimist@1.2.0, camelcase-keys@1.0.0, redent@1.0.0, read-pkg-up@1.0.1, normalize-package-data@2.3.4)
├── update-notifier@0.5.0 (is-npm@1.0.0, string-length@1.0.1, chalk@1.1.1, semver-diff@2.0.0, repeating@1.1.3, configstore@1.2.1, latest-version@1.0.1)
└── trash@3.0.0 (win-trash@2.0.0, osx-trash@2.0.0, xdg-trash@2.0.2)
previous command took 7s.
```

*Hmm, it's __still__ telling me I should be using `trash`. But I just tried that and it didn't work.*

```
 ❯ trash node_modules
 ❯ 
```

*__Unexpected success!__*

Is there a better way to handle upgrades to make it more clear to users what needs to be done?

I'm guessing most users don't know about `changelog` and might not take the time to look through the git history.

If you think this is an issue with [update-notifier](https://github.com/yeoman/update-notifier) I can create a new feature request there instead.

**Comment by LinusU** [comment]
I ran into this as well, quite frustrating...

Maybe if update-notifier could get support for outputting the new name. But then again, I guess that this doesn't happen all that often...

**Comment by sindresorhus** [comment]
Sorry about that. I was working on moving those modules into the main `trash` module so I deprecated them. Unfortunately right before publishing a new `trash` version I got interrupted IRL...

As for the confusion with `trash` becoming `trash-cli`. I think it would be useful to somehow have the ability to add a message to `update-notifier`. See https://github.com/yeoman/update-notifier/issues/56.

> I'm guessing most users don't know about changelog and might not take the time to look through the git history.

I know they don't, but it should be expected, as a major means something is breaking. The fact that people blindly update (mostly for non-CLI modules) is a scary thought.
